ocs_ci/utility/kms.py

```diff
@@ -308,20 +334,37 @@ def create_ocs_vault_resources(self):
         if not config.ENV_DATA.get("VAULT_SKIP_VERIFY"):
             self.create_ocs_vault_cert_resources()
 
-        # create oc resource secret for token
-        token_data = templating.load_yaml(constants.EXTERNAL_VAULT_KMS_TOKEN)
-        # token has to base64 encoded (with padding)
-        token_data["data"]["token"] = base64.b64encode(
-            # encode() because b64encode expects a byte type
-            self.vault_path_token.encode()
-        ).decode()  # decode() because b64encode returns a byte type
-        self.create_resource(token_data, prefix="token")
+        # Create resource and configure kubernetes auth method
+        if config.ENV_DATA.get("VAULT_AUTH_METHOD") == constants.VAULT_KUBERNETES_AUTH:
+            self.vault_auth_method = constants.VAULT_KUBERNETES_AUTH
+            self.create_ocs_kube_auth_resources()
+            self.vault_kube_auth_setup(token_reviewer_name=self.vault_cwd_kms_sa_name)
+            self.create_vault_kube_auth_role(
+                namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+                role_name=self.vault_kube_auth_role,
+                sa_name="rook-ceph-system,rook-ceph-osd,noobaa",
+            )
+            self.create_vault_kube_auth_role(
+                namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+                role_name="odf-rook-ceph-osd",
+                sa_name="rook-ceph-osd",
+            )
+        else:
+            # create oc resource secret for token
+            token_data = templating.load_yaml(constants.EXTERNAL_VAULT_KMS_TOKEN)
+            # token has to base64 encoded (with padding)
+            token_data["data"]["token"] = base64.b64encode(
+                # encode() because b64encode expects a byte type
+                self.vault_path_token.encode()
+            ).decode()  # decode() because b64encode returns a byte type
```

Comment:
can we use https://github.com/red-hat-storage/ocs-ci/blob/c62502258566c5792502ce55d654ece9a821faaa/ocs_ci/utility/utils.py#L3607

Reply:
@rachael-george You can make changes in followup PR

Reply:
Ack. Will do!